src/eos-updater-poll.c

```diff
@@ -17,199 +18,279 @@
+          *out_group_name = g_strdup_printf ("Source \"%s\"",
+                                             download_source_to_string (source));
+          return TRUE;
         }
-      else /* no size data available (may or may not be an error) */
+    }
+
+  *out_group_name = NULL;
+  return FALSE;
+}
+
+static gboolean
+read_config (SourcesConfig *sources_config,
+             GError **error)
+{
+  g_autoptr(GKeyFile) config = g_key_file_new ();
+  g_autofree gchar *config_file_path = get_config_file_path ();
+  g_auto(GStrv) download_order_strv = NULL;
+  g_autoptr(GError) local_error = NULL;
+  g_autofree gchar *group_name = NULL;
+
+  if (!g_key_file_load_from_file (config, config_file_path, G_KEY_FILE_NONE, &local_error))
+    {
+      EosUpdaterDownloadSource main_source = EOS_UPDATER_DOWNLOAD_MAIN;
+      /* The documentation is not very clear about which error is
+       * returned when the file is not found.
+       */
+      if (!g_error_matches (local_error, G_FILE_ERROR, G_FILE_ERROR_NOENT) &&
+          !g_error_matches (local_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_NOT_FOUND))
         {
-          eos_updater_set_full_download_size (updater, -1);
-          eos_updater_set_full_unpacked_size (updater, -1);
-          eos_updater_set_download_size (updater, -1);
-          eos_updater_set_unpacked_size (updater, -1);
-          eos_updater_set_downloaded_bytes (updater, -1);
-
-          /* shouldn't actually stop us offering an update, as long
-           * as the branch itself is resolvable in the next step,
-           * but log it anyway.
-           */
-          if (error)
-            {
-              message ("No size summary data: %s", error->message);
-              g_clear_error (&error);
-            }
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return FALSE;
         }
+
+      /* Config file was not found, fall back to the defaults
+       */
+      sources_config->download_order = g_array_sized_new (FALSE, /* not null terminated */
+                                                          FALSE, /* no clearing */
+                                                          sizeof (EosUpdaterDownloadSource),
+                                                          1);
+      g_array_append_val (sources_config->download_order, main_source);
+      sources_config->volume_path = NULL;
+
+      return TRUE;
     }
-  else /* csum == NULL means OnHold=true, nothing to do here */
-    eos_updater_set_state_changed (updater, EOS_UPDATER_STATE_READY);
 
- out:
-  if (error)
+  download_order_strv = g_key_file_get_string_list (config,
+                                                    DOWNLOAD_GROUP,
+                                                    ORDER_KEY,
+                                                    NULL,
+                                                    error);
+  if (download_order_strv == NULL)
+    return FALSE;
```

Comment:
I think it should error out here, since the user has provided an invalid configuration. With c29741b, there is now an example configuration file installed in `/usr/share/eos-updater/etc/eos-updater.conf` which the user can copy, and which is loaded if no configuration file exists in `/etc`, but not if loading an existing file fails.